### PR TITLE
Set max timeout for e2e tests to 20m per job

### PR DESCRIPTION
**Pull request by mcmire**
## **Description**

Sometimes e2e jobs will hang with the following error:

    Fontconfig error: Cannot load default config file: No such file: (null)

This commit sets the maximum time for a single e2e job to 20 minutes. We use the `timeout` executable to accomplish this. As a result, if running e2e tests takes longer than the allotted time, the build will fail with exit code 124.

## **Related issues**

Progresses: https://github.com/MetaMask/metamask-extension/issues/22564

## **Manual testing steps**

There isn't a way to test this unfortunately as the above error occurs intermittently. However, this commit should only affect CI.

## **Screenshots/Recordings**

(None)

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by mcmire** [comment]
As an alternative to using `timeout`, we could also use the CircleCI API to cancel the workflow that's being run, and this would allow us to print a message like "Build reached max time of 20 minutes" or something. But this would require adding an API token to the settings and I'm not sure if we wanted to do that or not: https://support.circleci.com/hc/en-us/articles/4411204604059-Automatically-cancel-build-after-set-amount-of-time

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Comment by Gudahtt** [comment]
I suspect we'll be creating a CircleCI token soon anyway for automatically collecting flaky test metrics. But this approach looks good to me, we can switch it out for the other strategy later if we need to.

File: .circleci/config.yml [hunk 1]
```diff
@@ -907,9 +907,11 @@ jobs:
           command: |
             if .circleci/scripts/test-run-e2e.sh
             then
-              MULTICHAIN=1 yarn test:e2e:chrome --retries 2 --debug
+              timeout 20m yarn test:e2e:chrome --retries 2 --debug
             fi
           no_output_timeout: 5m
+          environment:
+            MULTICHAIN: 1
```
Comment (HowardBraham) [inline .circleci/config.yml:914]: Oh cool you moved this to `environment`, that trick would have been useful on some things I tried in the past

**Review by HowardBraham — APPROVED** [review]

**Pull request merged**